### PR TITLE
python3Packages.apted: init at 1.0.3

### PR DESCRIPTION
**Pull request by McSinyx**
## Description of changes

[APTED](https://pypi.org/project/apted) is a Python library implementing the tree editing distance algorithm of the same name.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
@ofborg build python312Packages.apted

File: pkgs/development/python-modules/apted/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, fetchFromGitHub, unittestCheckHook }:
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/apted/default.nix:1]: Please format with the new formatter.

**Review by FliegendeWurst — DISMISSED** [review]

File: pkgs/development/python-modules/apted/default.nix (hunk 1 above)
Comment (McSinyx) [reply pkgs/development/python-modules/apted/default.nix]: Done!

**Comment by FliegendeWurst** [comment]
I think using `pyproject = true` would be preferred
```diff
diff --git a/pkgs/development/python-modules/apted/default.nix b/pkgs/development/python-modules/apted/default.nix
index 9a5a9eb979a7..617b02719872 100644
--- a/pkgs/development/python-modules/apted/default.nix
+++ b/pkgs/development/python-modules/apted/default.nix
@@ -3,12 +3,13 @@
   buildPythonPackage,
   fetchFromGitHub,
   unittestCheckHook,
+  setuptools,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage {
   pname = "apted";
   version = "1.0.3";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "JoaoFelipe";
@@ -17,6 +18,10 @@ buildPythonPackage rec {
     hash = "sha256-h8vJDC5TPpyhDxm1sHiXPegPB2eorEgyrNqzQOzSge8=";
   };
 
+  build-system = [
+    setuptools
+  ];
+
   nativeCheckInputs = [ unittestCheckHook ];
 
   meta = with lib; {
```

File: pkgs/development/python-modules/apted/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "JoaoFelipe";
+    repo = "apted";
+    rev = "828b3e3f4c053f7d35f0b55b0d5597e8041719ac";
+    hash = "sha256-h8vJDC5TPpyhDxm1sHiXPegPB2eorEgyrNqzQOzSge8=";
+  };
+
+  nativeCheckInputs = [ unittestCheckHook ];
+
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (McSinyx) [reply pkgs/development/python-modules/apted/default.nix]: Done!

File: pkgs/development/python-modules/apted/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  unittestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "apted";
+  version = "1.0.3";
+  format = "setuptools";
```
[the remark this answers is not in the document]
Comment (McSinyx) [reply pkgs/development/python-modules/apted/default.nix]: Done!  Cc: @FliegendeWurst

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/development/python-modules/apted/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "JoaoFelipe";
+    repo = "apted";
+    rev = "828b3e3f4c053f7d35f0b55b0d5597e8041719ac";
+    hash = "sha256-h8vJDC5TPpyhDxm1sHiXPegPB2eorEgyrNqzQOzSge8=";
+  };
+
+  build-system = [ setuptools ];
+  nativeCheckInputs = [ unittestCheckHook ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/apted/default.nix:23]: Could you please add `pythonImportsCheck` for a mock test?
Besides this LGTM.

Comment (McSinyx) [reply pkgs/development/python-modules/apted/default.nix:23]: Thanks, I've added the import check.

**Comment by natsukium** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 277506`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.apted</li>
    <li>python313Packages.apted</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.apted</li>
    <li>python313Packages.apted</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.apted</li>
    <li>python313Packages.apted</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.apted</li>
    <li>python313Packages.apted</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thank you.

**Pull request merged**